### PR TITLE
Product list

### PR DESCRIPTION
**Pull request by harshitpaliwal95**
- Add Product list page and Filters
- filter
1. high to low and low to high by price
2. high to low and low to high rating
3. filter by category
4. filter with rating slider

https://user-images.githubusercontent.com/60725346/159983113-84e87efd-08a9-48ab-979f-b2ac4b091e44.mov

### CODE REVIEW

**Review by agrawalshanu708 — COMMENTED** [review]
good to merge

File: src/pages/homePage/home.js [hunk 1]
```diff
@@ -36,13 +35,13 @@ export function Home() {
           </div>
           {/* <!-- popular-card --> */}
           <div className="popular-card">
+            {/* <ProductCard />
             <ProductCard />
             <ProductCard />
             <ProductCard />
```
Comment (sakshi006) [inline src/pages/homePage/home.js]: Remove the commented code

File: src/pages/productPage/componets/filter/filter.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+        <div className="check-list-item">
+          <input
+            name="checkbox"
+            type="checkbox"
+            checked={women}
+            onChange={(e) => {
+              dispatch({ type: "Women", payload: e.target.checked });
+            }}
+          />{" "}
```
Comment (sakshi006) [inline src/pages/productPage/componets/filter/filter.js]: You can remove these extra blankspaces

File: src/pages/productPage/componets/filter/filter.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        </div>
+        <div className="check-list-item">
+          <input
+            name="checkbox"
+            type="checkbox"
+            checked={kid}
+            onChange={(e) => {
+              dispatch({ type: "Kid", payload: e.target.checked });
+            }}
+          />{" "}
```
Comment (sakshi006) [inline src/pages/productPage/componets/filter/filter.js]: Here also

File: src/pages/productPage/product.js [hunk 4]
```diff
@@ -1,10 +1,51 @@
...
+      } catch (error) {
+        console.log(error.message);
+      }
+    })();
+  }, []);
+
+  const defaultData = [...product];
+  const newSortdata = sortData(defaultData, state.sortBy);
+  const newCategory = filterByCategory(newSortdata, state.category);
+  const filterData = latestProduct(newCategory, state.category);
```
Comment (sakshi006) [inline src/pages/productPage/product.js:33]: Here you can use the concept of composition of functions and use it here
For example: [see this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#function_composition_enabling_piping)

**Review by sakshi006 — APPROVED** [review]
The file structure and naming conventions are good.
made some comments, please look into that.

File: src/pages/homePage/home.js (hunk 1 above)
Comment (harshitpaliwal95) [reply src/pages/homePage/home.js]: its resolve now

File: src/pages/productPage/product.js (hunk 4 above)
Comment (harshitpaliwal95) [reply src/pages/productPage/product.js:33]: thanks for the suggestion ill look into it and try to implement it

**Pull request merged**